From b403ddd709578d11f1e6d32abfc94701f57cae16 Mon Sep 17 00:00:00 2001
From: Paul Kocialkowski <contact@paulk.fr>
Date: Thu, 25 Jan 2018 17:31:55 -0500
Subject: [PATCH] Avoid using git submodules for 3rdparty

This gets rid of git submodules entirely, to avoid the nuisance caused
by automatically checking them out.
---
 .gitmodules                   | 20 --------------------
 3rdparty/arm-trusted-firmware |  1 -
 3rdparty/blobs                |  1 -
 3rdparty/chromeec             |  1 -
 3rdparty/libgfxinit           |  1 -
 3rdparty/libhwbase            |  1 -
 3rdparty/vboot                |  1 -
 Makefile.inc                  | 12 ------------
 8 files changed, 38 deletions(-)
 delete mode 160000 3rdparty/arm-trusted-firmware
 delete mode 160000 3rdparty/blobs
 delete mode 160000 3rdparty/chromeec
 delete mode 160000 3rdparty/libgfxinit
 delete mode 160000 3rdparty/libhwbase
 delete mode 160000 3rdparty/vboot

diff --git a/.gitmodules b/.gitmodules
index c3270e6..3a617c7 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,23 +1,3 @@
-[submodule "3rdparty/blobs"]
-	path = 3rdparty/blobs
-	url = ../blobs.git
-	update = none
-	ignore = dirty
 [submodule "util/nvidia-cbootimage"]
 	path = util/nvidia/cbootimage
 	url = ../nvidia-cbootimage.git
-[submodule "vboot"]
-	path = 3rdparty/vboot
-	url = ../vboot.git
-[submodule "arm-trusted-firmware"]
-	path = 3rdparty/arm-trusted-firmware
-	url = ../arm-trusted-firmware.git
-[submodule "3rdparty/chromeec"]
-	path = 3rdparty/chromeec
-	url = ../chrome-ec.git
-[submodule "libhwbase"]
-	path = 3rdparty/libhwbase
-	url = ../libhwbase.git
-[submodule "libgfxinit"]
-	path = 3rdparty/libgfxinit
-	url = ../libgfxinit.git
diff --git a/3rdparty/arm-trusted-firmware b/3rdparty/arm-trusted-firmware
deleted file mode 160000
index b118723..0000000
--- a/3rdparty/arm-trusted-firmware
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit b1187232fdf819586ba8c8ece4a27a7515cbdc6d
diff --git a/3rdparty/blobs b/3rdparty/blobs
deleted file mode 160000
index 8eb92ba..0000000
--- a/3rdparty/blobs
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit 8eb92ba947e171df11b3c62f5f257ce69b9e2d55
diff --git a/3rdparty/chromeec b/3rdparty/chromeec
deleted file mode 160000
index 9fb1038..0000000
--- a/3rdparty/chromeec
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit 9fb10386a720d270e37ce61da3ff3a6d5a69951e
diff --git a/3rdparty/libgfxinit b/3rdparty/libgfxinit
deleted file mode 160000
index 42fb2d0..0000000
--- a/3rdparty/libgfxinit
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit 42fb2d065d604eb08c723ac6b96aeebb4c84cbd3
diff --git a/3rdparty/libhwbase b/3rdparty/libhwbase
deleted file mode 160000
index 6685971..0000000
--- a/3rdparty/libhwbase
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit 66859712e4817288591908d737dbf41ddea31c3a
diff --git a/3rdparty/vboot b/3rdparty/vboot
deleted file mode 160000
index f6780a3..0000000
--- a/3rdparty/vboot
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit f6780a36ff19b36abcdb5ace903c4ae2272fb574
diff --git a/Makefile.inc b/Makefile.inc
index 413f7ad..ed2d839 100644
--- a/Makefile.inc
+++ b/Makefile.inc
@@ -191,18 +191,6 @@ ifeq ($(CONFIG_COVERAGE),y)
 ramstage-c-ccopts += -fprofile-arcs -ftest-coverage
 endif
 
-ifneq ($(UPDATED_SUBMODULES),1)
-# try to fetch non-optional submodules if the source is under git
-forgetthis:=$(if $(GIT),$(shell git submodule update --init))
-ifeq ($(CONFIG_USE_BLOBS),y)
-# this is necessary because 3rdparty/blobs is update=none, and so is ignored
-# unless explicitly requested and enabled through --checkout
-forgetthis:=$(if $(GIT),$(shell git submodule update --init --checkout 3rdparty/blobs))
-endif
-UPDATED_SUBMODULES:=1
-COREBOOT_EXPORTS += UPDATED_SUBMODULES
-endif
-
 postcar-c-deps:=$$(OPTION_TABLE_H)
 ramstage-c-deps:=$$(OPTION_TABLE_H)
 romstage-c-deps:=$$(OPTION_TABLE_H)
-- 
1.9.1

